arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -1522,16 +1536,24 @@ Result RestReplicationHandler::processRestoreCoordinatorAnalyzersBatch() {
 /// by key
 ////////////////////////////////////////////////////////////////////////////////
 
-Result RestReplicationHandler::processRestoreUsersBatch(std::string const& collectionName) {
-  std::unordered_map<std::string, VPackValueLength> latest;
-  VPackBuilder allMarkers;
+Result RestReplicationHandler::processRestoreUsersBatch(bool generateNewRevisionIds) {
+  VPackBuilder documentsToInsert;
 
-  Result res = parseBatch(collectionName, latest, allMarkers);
-  if (res.fail()) {
-    return res;
-  }
+  {
+    // this "fake" transaction here is only needed to get access to the underlying
+    // _users collection. we will not write anything into the collection here
+    auto ctx = transaction::StandaloneContext::Create(_vocbase);
+    SingleCollectionTransaction trx(ctx, StaticStrings::UsersCollection, AccessMode::Type::READ);
 
-  VPackSlice allMarkersSlice = allMarkers.slice();
+    Result res = trx.begin();
+    if (res.ok()) {
+      res = parseBatch(trx, StaticStrings::UsersCollection, documentsToInsert, generateNewRevisionIds);
+    }
+    if (res.fail()) {
+      return res;
+    }
```

Comment:
These lines are identical to those in `processRestoreCoordinatorAnalyzersBatch` (except for the collection name). Can we keep it DRY and extract them into a separate function?

Reply:
Good catch. Done.